src/sched/lthread.c

```diff
@@ -63,7 +63,33 @@ static void lthread_rundestructors(struct lthread* lt);
 static void dummy_0()
 {
 }
-weak_alias(dummy_0, __do_orphaned_stdio_locks);
+// weak_alias(dummy_0, __do_orphaned_stdio_locks);
+
+static int spawned_ethreads = 1;
+
+static int __init_ethread_tp(void *p)
+{
+	struct schedctx *td = p;
+
+	td->self = td;
+	// Prevent collisions with lthread TIDs which are assigned to newly spawned
+	// lthreads incrementally, starting from one.
+	td->tid = INT_MAX - a_fetch_add(&spawned_ethreads, 1);
+	//td->locale = &libc.global_locale;
+	//td->robust_list.head = &td->robust_list.head;
+	//libc.can_do_threads = 1;
+	return 0;
+}
+
+/* Initialisation of ethread/scheduler TLS */
+void init_ethread_tls()
+{
+	void *mem = __scheduler_self();
+
+	/* Failure to initialize thread pointer is always fatal. */
+	if (__init_ethread_tp(mem) < 0)
+		a_crash();
```

Comment:
This is dead code, `__init_ethread_tp` unconditionally returns 0 so this is never reached.

It's probably a good idea to simply inline `__init_ethread_tp` here.